src/main/java/org/spongepowered/common/data/processor/common/SpawnerUtils.java

```diff
@@ -0,0 +1,100 @@
+package org.spongepowered.common.data.processor.common;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.entity.passive.EntityPig;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.MobSpawnerBaseLogic;
+import net.minecraft.world.World;
+import org.spongepowered.api.entity.EntitySnapshot;
+import org.spongepowered.api.entity.EntityType;
+import org.spongepowered.api.entity.EntityTypes;
+import org.spongepowered.api.util.weighted.TableEntry;
+import org.spongepowered.api.util.weighted.WeightedObject;
+import org.spongepowered.api.util.weighted.WeightedSerializableObject;
+import org.spongepowered.api.util.weighted.WeightedTable;
+import org.spongepowered.common.entity.SpongeEntitySnapshot;
+import org.spongepowered.common.entity.SpongeEntitySnapshotBuilder;
+import org.spongepowered.common.registry.type.entity.EntityTypeRegistryModule;
+
+import java.util.Optional;
+
+public class SpawnerUtils {
+
+    public static Optional<Entity> createEntity(EntityType type, World world) {
```

Comment:
I think that the method you've got right here will also have a default position of 0,0,0 so it's no worse. difference is that world.createEntity does validation on the position so if it returns a non-empty result you know it can spawn there